### ISSUE TITLE
Unable to create sites on MacOS

### ISSUE THREAD

**Issue by alhambrav** [craftercms/craftercms#7026]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [x] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When running CrafterCMS on MacOS with OpenSearch 2.13.0, the OpenSearch docker exits immediately during startup and users are then unable to create sites, etc.

### Steps to reproduce

Steps:

1. Clone, build, deploy and start CrafterCMS on MacOS
2. Open your Docker Desktop and look at the logs of the search container
3. See the message in the logs:
```
2024-07-09 12:47:09 OpenSearch 2.12.0 onwards, the OpenSearch Security Plugin a change that requires an initial password for 'admin' user. 
2024-07-09 12:47:09 Please define an environment variable 'OPENSEARCH_INITIAL_ADMIN_PASSWORD' with a strong password string. 
...
2024-07-09 12:47:15 No custom admin password found. Please provide a password via the environment variable OPENSEARCH_INITIAL_ADMIN_PASSWORD.
```
From the OpenSearch docs, we need to add the environment variable `OPENSEARCH_INITIAL_ADMIN_PASSWORD` when creating the docker for OpenSearch, and why the search container exits immediately during startup of CrafterCMS.

https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/

As a temporary fix, I was able to get OpenSearch running by adding `OPENSEARCH_INITIAL_ADMIN_PASSWORD` as mentioned in the Docker logs. I hard coded some password in the `crafter.sh` file, in the function `CreateOpenSearchDocker`:

```
function createOpenSearchDocker() {
  if docker inspect "$SEARCH_DOCKER_NAME" > /dev/null 2>&1; then
    cecho "Docker image $SEARCH_DOCKER_NAME found, unable to start. Please remove this image before starting.\n To remove the image, run: docker rm $SEARCH_DOCKER_NAME\n" "error"
    exit 21
  else
    docker create -p $SEARCH_PORT:9200 -e "discovery.type=single-node" -e "plugins.security.disabled=true" -e "OPENSEARCH_INITIAL_ADMIN_PASSWORD=SomeSecretPassword" -v "$SEARCH_INDEXES_DIR":/usr/share/opensearch/data/ --name "$SEARCH_DOCKER_NAME" opensearchproject/opensearch:2.13.0 > /dev/null 2>&1
  fi
}
```

### Relevant log output

```shell
2024-07-09 12:47:09 Enabling OpenSearch Security Plugin
2024-07-09 12:47:09 Enabling execution of install_demo_configuration.sh for OpenSearch Security Plugin 
2024-07-09 12:47:09 OpenSearch 2.12.0 onwards, the OpenSearch Security Plugin a change that requires an initial password for 'admin' user. 
2024-07-09 12:47:09 Please define an environment variable 'OPENSEARCH_INITIAL_ADMIN_PASSWORD' with a strong password string. 
2024-07-09 12:47:09 If a password is not provided, the setup will quit. 
2024-07-09 12:47:09  For more details, please visit: https://opensearch.org/docs/latest/install-and-configure/install-opensearch/docker/
2024-07-09 12:47:10 ### OpenSearch Security Demo Installer
2024-07-09 12:47:10 ### ** Warning: Do not use on production or public reachable systems **
2024-07-09 12:47:10 OpenSearch install type: rpm/deb on Linux 6.6.31-linuxkit amd64
2024-07-09 12:47:10 OpenSearch config dir: /usr/share/opensearch/config/
2024-07-09 12:47:10 OpenSearch config file: /usr/share/opensearch/config/opensearch.yml
2024-07-09 12:47:10 OpenSearch bin dir: /usr/share/opensearch/bin/
2024-07-09 12:47:10 OpenSearch plugins dir: /usr/share/opensearch/plugins/
2024-07-09 12:47:10 OpenSearch lib dir: /usr/share/opensearch/lib/
2024-07-09 12:47:10 Detected OpenSearch Version: 2.13.0
2024-07-09 12:47:10 Detected OpenSearch Security Version: 2.13.0.0
2024-07-09 12:47:15 No custom admin password found. Please provide a password via the environment variable OPENSEARCH_INITIAL_ADMIN_PASSWORD.
```

**Comment by alhambrav** [comment]
Verified fixed.

**Comment by alhambrav** [comment]
Re-opening to fix 4.1.x also.
@phuongnq Please add the fix for 4.1.x too. Thanks!

**Comment by alhambrav** [comment]
Verified fixed for 4.1.x
